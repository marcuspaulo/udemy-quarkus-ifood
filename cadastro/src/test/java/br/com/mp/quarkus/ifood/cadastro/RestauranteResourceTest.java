package br.com.mp.quarkus.ifood.cadastro;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;
import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import org.approvaltests.Approvals;
import org.junit.Test;

import static io.restassured.RestAssured.given;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(CadastroTestLifecycleManager.class)
public class RestauranteResourceTest {

    @Test
    @DataSet("restaurantes-cenario-1.yml")
    public void testBuscarRestautantes() {
        String resultado = given().log().all()
                .when()
                .get("/restaurantes")
                .then()
                .statusCode(200)
                .extract().asString();
        Approvals.verifyJson(resultado);
    }
}
