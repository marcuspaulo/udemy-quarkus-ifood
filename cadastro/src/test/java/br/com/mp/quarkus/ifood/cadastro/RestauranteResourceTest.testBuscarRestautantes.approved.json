[
  {
    "dataAtualizacao": "2021-03-26T21:35:39.92Z[UTC]",
    "dataCriacao": "2021-03-26T21:35:39.92Z[UTC]",
    "id": 2,
    "nome": "Restaurante",
    "proprietario": "string"
  },
  {
    "dataAtualizacao": "2021-03-26T21:36:22.308Z[UTC]",
    "dataCriacao": "2021-03-26T21:36:22.308Z[UTC]",
    "id": 3,
    "nome": "Restaurante Japonês",
    "proprietario": "Proprietario"
  },
  {
    "dataAtualizacao": "2021-03-26T21:37:15.689Z[UTC]",
    "dataCriacao": "2021-03-26T21:22:30.857Z[UTC]",
    "id": 1,
    "nome": "Restaurante Japonês Sushi",
    "proprietario": "string"
  },
  {
    "dataAtualizacao": "2021-03-26T21:43:43.984Z[UTC]",
    "dataCriacao": "2021-03-26T21:43:43.984Z[UTC]",
    "id": 5,
    "nome": "Pizzaria",
    "proprietario": "string"
  }
]